status/versions.go

```diff
@@ -193,15 +193,46 @@ type p8sResponseVersion struct {
 	Revision string `json:"revision"`
 }
 
+type jaegerResponseVersion struct {
+	Version string `json:"gitVersion"`
+}
+
 func tracingVersion() (*ExternalServiceInfo, error) {
 	tracingConfig := config.Get().ExternalServices.Tracing
 
 	if !tracingConfig.Enabled {
 		return nil, nil
 	}
+
 	product := ExternalServiceInfo{}
 	product.Name = string(tracingConfig.Provider)
 	product.Url = tracingConfig.URL
+
+	if product.Url != "" {
+		// try to determine version by querying
+		if tracingConfig.Provider == config.JaegerProvider {
+			body, statusCode, _, err := httputil.HttpGet(product.Url, nil, 10*time.Second, nil, nil)
+			if err != nil || statusCode > 399 {
+				log.Infof("jaeger version check failed: url=[%v], code=[%v]", product.Url, statusCode)
+			} else {
+				// Jaeger does not provide api to get version, so it is obtained from js function inside html main page
+				// const JAEGER_VERSION = {"gitCommit: xxx, gitVersion: yyy, buildDate: zzz"}
```

Comment:
Tricky!  I never would have thought of that.  I looked all around the API but couldn't find a way.  I also looked to see if the service was labeled with a version (the istio components are, the grafana component is.. but not jaeger).